### PR TITLE
Added proof-of-concept for Force Refresh feature extension to .NET Core/FX SqlClient driver

### PR DESCRIPTION
**Pull request by shuai626**

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/SqlClient?pullRequest=1176) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/SqlClient?pullRequest=1176) before we can accept your contribution.<br/><br/>:x: shuai626 [sign now](https://cla.dotnetfoundation.org/dotnet/SqlClient?pullRequest=1176)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/SqlClient?pullRequest=1176) it.</sub>

**Review by David-Engel — APPROVED** [review]

**Review by cheenamalhotra — APPROVED** [review]

**Comment by ErikEJ** [comment]
What problem does this PR solve?

**Comment by cheenamalhotra** [comment]
It's a POC work in progress in another branch (not in main right now), it is aimed to enhance Azure connection recovery by forcing refresh on server side in event of transient error.

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsEnums.cs [hunk 1]
```diff
@@ -228,7 +229,8 @@ public enum FeatureExtension : uint
             AzureSQLSupport = 1 << (TdsEnums.FEATUREEXT_AZURESQLSUPPORT - 1),
             DataClassification = 1 << (TdsEnums.FEATUREEXT_DATACLASSIFICATION - 1),
             UTF8Support = 1 << (TdsEnums.FEATUREEXT_UTF8SUPPORT - 1),
-            SQLDNSCaching = 1 << (TdsEnums.FEATUREEXT_SQLDNSCACHING - 1)
+            SQLDNSCaching = 1 << (TdsEnums.FEATUREEXT_SQLDNSCACHING - 1),
+            ForceRefresh = 1 << (TdsEnums.FEATUREEXT_FORCEREFRESH - 1)
```
Comment (ErikEJ) [inline src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsEnums.cs:233]: Suggest ending with , to make future PRs more readable

**Pull request merged**